src/metabase/upload.clj

```diff
@@ -181,18 +186,16 @@
 
 (defn- is-pk?
   [[col-name type]]
-  (and (#{"id" "pk"} (u/lower-case-en (name col-name)))
-       (isa? type ::int)))
+  (and (#{"id" "pk" "uuid" "guid"} (u/lower-case-en (name col-name)))
+       ((set (keys base-type->pk-type)) type)))
```

Comment:
for readability maybe define `base-types` separately, like `types`

Reply:
Fair enough, done